### DISCUSSION TITLE
Shortcode does not work in html tables in version 1.4

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#8778] [Q&A]
### Description

The table looks like this in quarto 1.3.450

![Screenshot 2024-02-18 at 23 31 23](https://github.com/quarto-dev/quarto-cli/assets/5083500/2c836cfa-628f-46be-b25f-19b05bf541aa)

and like this in quarto 1.4.549

![Screenshot 2024-02-18 at 23 31 37](https://github.com/quarto-dev/quarto-cli/assets/5083500/7a9a86ec-a193-4e69-948f-dddb729b0a60)

My qmd file looks like this

````
---
title: Table with shortcode
format: html
filters:
  - fontawesome
---

```{r}
dfr <- data.frame(x="a",y="{{< fa calendar >}}")
htmlTable::htmlTable(dfr)
```
````

```
R 4.3.0
htmlTable_2.4.1
quarto fontawesome extension: 1.1.0
```

**Comment by cscheid** [comment]
That's [by design](https://quarto.org/docs/prerelease/1.4/ast.html). Unless the output from the HTML table is specifically marked for us to process it as markdown, we interpret the output as html. `{{< fa calendar >}}` is valid HTML text, so that's what we do.

**Reply by cscheid** [reply]
(somewhat counter-intuitively, you can use `html-table-processing: none` in your front matter; that causes Quarto _not_ to parse the HTML table into a Table AST element, and then the shortcode will be evaluated from within the raw block itself.)

**Reply by royfrancis** [reply]
Thanks! But that doesn't seem to make a difference. Perhaps i am using it incorrectly.

For my real use case, I have raw HTML as well as shortcodes in my table. So I guess I want quarto to convert the shortcodes into html first. So, I am not sure if disabling table processing is the right option.

Now I have three documents: q1.3, q1.4, q1.4 with no table processing. I also added a few more chunks with a few other table packages for comparison. These are the chunks in order: markdown table, default R table output, htmlTable, kable, paged and gt.

![Screenshot 2024-02-19 at 09 38 56](https://github.com/quarto-dev/quarto-cli/assets/5083500/788a7437-e4ee-4056-bbf8-f1f3e3a10c63)

Here is one of the qmd files:

````
---
title: q1.4 no processing
format:
  html:
    html-table-processing: none
filters:
  - fontawesome
---

## Markdown table

|x|y|z|
|---|---|---|
|a|{{< fa calendar >}}|<span style="color:red">styled</span>|


## Default R table

```{r}
dfr <- data.frame(x="a",y="{{< fa calendar >}}",z='<span style="color:red">styled</span>')
dfr
```

## htmlTable

```{r}
htmlTable::htmlTable(dfr)
```

## kable

```{r}
knitr::kable(dfr)
```

## paged

```{r}
rmarkdown::paged_table(dfr)
```

## gt

```{r}
gt::gt(dfr)
```
````

**Reply by mcanouil** [reply]
The issue here is that you output HTML but want somehow markdown in it to be parsed without telling Quarto there is markdown within the HTML.

Take a look at what you are producing using `keep-md: true`.

````qmd
---
title: q1.4 no processing
format: html
---

## Markdown table

|x|y|z|
|---|---|---|
|a|{{< meta title >}}|`<span style="color:red">styled</span>`{=html}|


## Default R table

```{r}
dfr <- data.frame(x="a",y="{{< meta title >}}",z="<span style='color:red;'>styled</span>")
dfr
```

## gt

```{r}
gt::gt(dfr) |>
  gt::fmt_markdown()
```

```{r}
#| html-table-processing: none
gt::gt(dfr) |>
  gt::fmt_markdown()
```
````

<img width="691" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/2db63f64-7514-4c9b-b7e3-2ab8e6ffe017">
